src/main/java/com/arrl/radiocraft/client/render/BlinkingAutoGlowingGeoLayer.java

```diff
@@ -0,0 +1,118 @@
+            Level level = be.getLevel();
+            if (level == null) return;
+
+            // Check for radio in the charger slot
+            var stack = be.inventory.getStackInSlot(0);
+            if (stack == null || stack.isEmpty()) {
+                // No radio present - LED is off (dark)
+                redTint = greenTint = blueTint = 0.0f;
+            } else {
+                var radioEnergy = stack.getCapability(net.neoforged.neoforge.capabilities.Capabilities.EnergyStorage.ITEM);
+                var blockEnergy = be.energyStorage;
+
+                if (radioEnergy != null && blockEnergy != null) {
+                    int radioStored = radioEnergy.getEnergyStored();
+                    int radioMax = radioEnergy.getMaxEnergyStored();
+                    int blockStored = blockEnergy.getEnergyStored();
+
+                    // LED off if radio or charger has no energy
+                    if (radioStored <= 0 || blockStored <= 0) {
+                        redTint = greenTint = blueTint = 0.0f;
+                    } else {
+                        // Interpolate LED color from red (0% charge) to green (100% charge)
+                        float charge = (float) radioStored / (float) Math.max(1, radioMax);
+                        redTint = 1.0f - charge;
+                        greenTint = charge;
+                        blueTint = 0.0f;
+
+                        // Apply blinking effect while charging
+                        boolean charging = radioStored < radioMax && (blockStored > 0 || be.isInfinite());
+
+                        if (charging) {
+                            long t = level.getGameTime();
+                            int period = 8; // Blink cycle duration in ticks
+                            int onFor = 4;  // Ticks LED is on per cycle
+                            boolean lightOn = (t % period) < onFor;
+                            
+                            if (!lightOn) {
+                                // During "off" phase, render very dark gray instead of pure black.
+                                // Pure black (0,0,0) is treated as transparent by some render types.
+                                redTint = greenTint = blueTint = 0.02f;
+                            }
+                        }
+                        // When fully charged or not charging: steady on (no blinking)
+                    }
+                }
+            }
+        }
+
+        // Convert float RGB values to packed integer (ARGB format)
+        int r = Math.max(0, Math.min(255, (int) (redTint * 255.0f)));
+        int g = Math.max(0, Math.min(255, (int) (greenTint * 255.0f)));
+        int b = Math.max(0, Math.min(255, (int) (blueTint * 255.0f)));
+        int colour = (255 << 24) | (r << 16) | (g << 8) | b;
+
+        // Use fullbright lighting so LED color appears consistent regardless of ambient light
+        int fullBright = 15728880; // Packed light value for full brightness (15,15)
```

Comment:
Out of curiosity, how was this value generated? Is it a pair of bit-shifted values?

Reply:
And actually, I don't mind the hex equivelant here; it makes the value more readable compared to the current integer representation (ie, I prefer `0xF000F0` if Java lets you put that verbatim)